be/src/storage/olap_server.cpp

```diff
@@ -188,7 +166,7 @@ void* StorageEngine::_base_compaction_thread_callback(void* arg, DataDir* data_d
     //string last_base_compaction_fs;
     //TTabletId last_base_compaction_tablet_id = -1;
     Status status = Status::OK();
-    while (!_stop_bg_worker) {
+    while (!_bg_worker_stopped) {
```

Comment:
Since you have using the `std::memory_order_release` to write `_bg_worker_stopped`, you should read it by `std::memory_order_consume` all the times. Maybe you can call `StorageEngine::bg_worker_stopped()` in these places.